src/core/lib/gprpp/ref_counted.h

```diff
@@ -34,14 +34,58 @@
 
 namespace grpc_core {
 
+// PolymorphicRefCount enforces polymorphic destruction of RefCounted.
+class PolymorphicRefCount {
+ public:
+  GRPC_ABSTRACT_BASE_CLASS
```

Comment:
@markdroth 
There is a compilation failures on Kokoru which doesn't happen on my machine. I had to add the following macros to fix it after your review:
`GRPC_ABSTRACT_BASE_CLASS` and `GPRC_ALLOW_CLASS_TO_USE_NON_PUBLIC_DELETE`

PTAL. Thank you

Reply:
LGTM

Reply:
Thank you!